packages/app-check/src/internal-api.ts

```diff
@@ -108,7 +112,8 @@ export async function getToken(
    */
   try {
     if (state.customProvider) {
-      token = await state.customProvider.getToken();
+      const customToken = await state.customProvider.getToken();
+      token = { ...customToken, issuedAtTimeMillis: Date.now() };
```

Comment:
the customToken might be issued earlier than the current timestamp. It might be okay to assume it to be the current timestamp for the purpose of determining the refresh time, but is it the case? We should add comments to clarify.

Reply:
I might be misunderstanding but I think that if it was issued earlier, it would be in the cache and this function would have returned in line 105? Once it's fetched here it'll be written to storage in line 144. Unless the custom `getToken()` method has its own code to read/write from its own separate cache, where someone might have wiped Firebase AppCheck's cache but didn't wipe out the user's custom cache??, which, should we worry about that?

Reply:
It could happen when `customProvider.getToken()` executes for the first time before cache comes into play. Since `customProvider` is implemented by developers, it can return a token created 3 hours ago or 2 days ago.